test/functional/wallet-hd-restore.py

```diff
@@ -0,0 +1,162 @@
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.node_args = [['-usehd=0'], ['-usehd=1', '-keypool=100', '-hdignoregaplimit=0']]
+
+    def setup_network(self):
+        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir, self.node_args)
+        self.is_network_split = False
+        connect_nodes_bi(self.nodes, 0, 1)
+
+    def run_test (self):
+        tmpdir = self.options.tmpdir
+
+        print("Initialize wallet including backups of unencrypted and encrypted wallet")
+        # stop and backup original wallet (only keypool has been initialized)
+        self.stop_node(1)
+        shutil.copyfile(tmpdir + "/node1/regtest/wallet.dat", tmpdir + "/hd.bak")
+
+        # start again and encrypt wallet
+        self.nodes[1] = start_node(1, self.options.tmpdir, self.node_args[1])
+        self.nodes[1].encryptwallet('test')
+        bitcoind_processes[1].wait()
+        # node will be stopped during encryption, now do a backup
+        shutil.copyfile(tmpdir + "/node1/regtest/wallet.dat", tmpdir + "/hd.enc.bak") 
+
+        # start the node with encrypted wallet, get address in new pool at pos 50 (over the gap limit)
+        self.nodes[1] = start_node(1, self.options.tmpdir, self.node_args[1])
+        for _ in range(50):
+            addr_enc_oldpool = self.nodes[1].getnewaddress()
+
+        # now make sure we retrive an address in the extended pool
+        self.nodes[1].walletpassphrase("test", 10)
+        for _ in range(80):
+            addr_enc_extpool = self.nodes[1].getnewaddress()
+
+        # stop and load initial backup of the unencrypted wallet
+        self.stop_node(1)
+        os.remove(tmpdir + "/node1/regtest/wallet.dat")
+        shutil.copyfile(tmpdir + "/hd.bak", tmpdir + "/node1/regtest/wallet.dat")
+        self.nodes[1] = start_node(1, self.options.tmpdir, self.node_args[1])
+        connect_nodes_bi(self.nodes,0,1)
+        for _ in range(10):
+            addr = self.nodes[1].getnewaddress()
+
+        self.nodes[0].generate(101)
+        addr = self.nodes[1].getnewaddress()
+        assert_equal(self.nodes[1].validateaddress(addr)['hdkeypath'], "m/0'/0'/11'")
+
+        rawch = self.nodes[1].getrawchangeaddress()
+        txid = self.nodes[0].sendtoaddress(addr, 1)
+        n0addr = self.nodes[0].getnewaddress()
+        txdata = self.nodes[0].createrawtransaction([], {rawch : 2.0, n0addr: 3.0})
+
+        txdata_f = self.nodes[0].fundrawtransaction(txdata)
+        txdata_s = self.nodes[0].signrawtransaction(txdata_f['hex'])
+        txid = self.nodes[0].sendrawtransaction(txdata_s['hex'])
+
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        assert_equal(self.nodes[1].getbalance(), 3)
+        assert_equal(self.nodes[1].listtransactions()[0]['category'], "receive")
+
+        print("Testing with unencrypted wallet")
+        self.stop_node(1)
+        shutil.rmtree(tmpdir + "/node1/regtest")
```

Comment:
please don't do this! It erases the debug.log file. Are you able to be more targeted in what you're erasing?